src/ol/renderer/layerrenderer.js

```diff
@@ -191,7 +195,8 @@ ol.renderer.Layer.prototype.scheduleExpireCache =
              */
             function(tileSource, map, frameState) {
               var tileSourceKey = goog.getUid(tileSource).toString();
-              tileSource.expireCache(frameState.usedTiles[tileSourceKey]);
+              tileSource.expireCache(frameState.viewState.projection,
+                                     frameState.usedTiles[tileSourceKey]);
```

Comment:
> The projection parameter only specifies the projection of the usedTiles (and these are not removed from the cache even if it's over the high water mark).

Actually what your code does is: prune the cache for all projections that are not used (by passing `{}` as `usedTiles`), and expire the cache for the used projection (i.e. remove the least recently used tiles from the cache if it is over high water mark).